website/docs/guides/2.0-upgrade-guide.html.markdown

```diff
@@ -0,0 +1,348 @@
+
+### Resource: `azurerm_servicebus_queue`
+
+The deprecated `location` field will be removed, since this is no longer used.
+
+The deprecated `enable_batched_operations` field will be removed, since this is no longer used.
+
+The deprecated `support_ordering` field will be removed, since this is no longer used.
+
+### Resource: `azurerm_servicebus_subscription`
+
+The deprecated `location` field will be removed, since this is no longer used.
+
+The deprecated `dead_lettering_on_filter_evaluation_exceptions` field will be removed, since this is no longer used.
+
+### Resource: `azurerm_servicebus_topic`
+
+The deprecated `location` field will be removed, since this is no longer used.
+
+The deprecated `enable_filtering_messages_before_publishing` field will be removed, since this is no longer used.
+
+### Resource: `azurerm_storage_account`
+
+The deprecated `account_type` field will be removed. This has been split into the fields `account_tier` and `account_replication_type`.
+
+### Resource: `azurerm_subnet`
+
+The deprecated field `network_security_group_id` will be removed. This has been replaced by the `azurerm_subnet_network_security_group_association` resource.
+
+The deprecated field `route_table_id` will be removed. This has been replaced by the `azurerm_subnet_route_table_association` resource.
+
+### Resource: `azurerm_virtual_machine`
+
+The `azurerm_virtual_machine` resource will be deprecated in favour of two new resources: `azurerm_linux_virtual_machine` and `azurerm_windows_virtual_machine`.
+
+Splitting the Virtual Machine resource in two allows us to both provide finer-grain validation for this resource and update the schema.
+
+The existing `azurerm_virtual_machine` Resource will continue to be available in it's current form, however it will eventually be deprecated and we recommend using the new resources going forward.
+
+### Resource: `azurerm_virtual_machine_scale_set`
+
+The `azurerm_virtual_machine_scale_set` resource will be deprecated in favour of two new resources: `azurerm_linux_virtual_machine_scale_set` and `azurerm_windows_virtual_machine_scale_set`.
+
+Splitting the Virtual Machine Scale Set resource in two allows us to both provide finer-grain validation for this resource and update the schema.
+
+The existing `azurerm_virtual_machine_scale_set` Resource will continue to be available in it's current form, however it will eventually be deprecated and we recommend using the new resources going forward.
+
+---
+
+We've spent the past few months laying the groundwork for these changes - and whilst we appreciate that your Terraform Configurations may require code changes to upgrade to 2.0 - we take Semantic Versioning seriously and so try our best to limit these changes to major versions.
+
+### Pinning your Provider Version
```

Comment:
You may want to consider putting this information at the top so people can immediately see:
* How to consistently upgrade
* How to consistently prevent the upgrade